test-network: optionally remove state file of networkd

yuwata opened the pull request:
Otherwise, some tests may disturb others, e.g., NetworkdNetWorkTests.test_routing_policy_rule_port_range and NetworkdNetWorkTests.test_routing_policy_rule.

cc @ssahani.

ssahani commented:
Well it just a symlink. OK with me .
```
$ ll /var/run
lrwxrwxrwx. 1 root root 6 Apr 25  2018 /var/run -> ../run
```

yes better clear all the state files by default. we just need a fresh for each test case.

ssahani commented on test/test-network/systemd-networkd-tests.py:
```diff
@@ -117,8 +118,13 @@ def remove_log_file(self):
+            subprocess.check_call('systemctl stop systemd-networkd', shell=True)
+            os.remove(os.path.join(networkd_runtime_directory, 'state'))
+            subprocess.check_call('systemctl start systemd-networkd', shell=True)
+        else:
```
even for LLDP and DHCP should not we clean everything ? I think this this should be by default.

yuwata replied:
I am not sure, but relevant links are always removed. So, I expect it is safe.
If these files disturbs tests, then let's remove them on another PR.

yuwata commented:
@ssahani Thank you for the comments.
Updated. Now state file is always removed. I also added a commit to suppress warning messages about socket unit.

ssahani commented:
LGTM
